### PR TITLE
Reconnection fixes D-01259

### PR DESCRIPTION
**Pull request by timotree3**
- Increases max_reconnects to envoy from 5 to unlimited.
- Emits `disconnected` signals at appropriate times so that EC UI can display a message after 3 failed reconnects.

### CODE REVIEW

**Comment by timotree3** [comment]
The integration tests are failing and I don't understand why. They're timing out in `Chaperone.disconnect()` during `Chaperone.signUp()`. I tried logging `this.conn` in chaperone and it was `undefined`. The unit tests are passing and I confirmed that the fix worked on a chaperone using a real holoport. Is this the problem with chaperone develop mode you were referring to earlier? @JettTech

File: src/index.ts [hunk 1]
```diff
@@ -710,19 +710,35 @@ class Chaperone {
...
+      clearTimeout(disconnectedTimer)
+      disconnectedTimer = setTimeout(this.alertDisconnected.bind(this), reconnect_interval * 3)
     })
     this.conn.on("error", (err) => {
       log("WEBSOCKET TO ENVOY HAS ERROR. (%s) Ready state:", err, this.websocket()?.readyState)
     })
     await this.conn.opened(timeout);
     this.opened = true;
 
+    this.alertParent("connected");
```
Comment (JettTech) [inline src/index.ts:735]: Can we use `alertConnected()` here instead, to keep it consistent?

File: src/index.ts [hunk 2]
```diff
@@ -814,6 +829,7 @@ class Chaperone {
         await this.signOut();
         throw err;
       }
+      this.alertParent("connected");
```
Comment (JettTech) [inline src/index.ts:832]: Same as above - let's use `alertConnected()` here instead.

File: src/index.ts [hunk 3]
```diff
@@ -710,19 +710,35 @@ class Chaperone {
...
+    this.conn = new Client(ws_url, {
+      autoconnect: true,
+      reconnect: true,
+      reconnect_interval,
+      max_reconnects: 0,
+    });
     this.conn.on("close", () => {
       log("WEBSOCKET TO ENVOY IS CLOSING. Ready state:", this.websocket()?.readyState)
+      clearTimeout(disconnectedTimer)
+      disconnectedTimer = setTimeout(this.alertDisconnected.bind(this), reconnect_interval * 3)
```
Comment (JettTech) [inline src/index.ts:727]: Why a multiple of 3?  (Looks fine btw, just wondering what the reasoning was behind this.)

**Review by JettTech — APPROVED** [review]
Left a few comments, but overall all tests now pass and it looks good.

Just update with latest develop and I think we'll be good to go! :rocket:

File: src/index.ts (hunk 3 above)
Comment (timotree3) [reply src/index.ts:728]: just thought we should give a couple of tries first. It's arbitrary

**Pull request merged**